File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;

        array<char, 26> count{};
        for (int i = 0; i < s.size(); i++) {
            count.at(s.at(i) - 'a')++;
            count.at(t.at(i) - 'a')--;
        }

        for (const int& num : count) {
            if (num != 0) return false;
        }

        return true;
    }
};

File: Data Structures & Algorithms/reverse-nodes-in-k-group/submission-2.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
    public ListNode ReverseKGroup(ListNode head, int k) {
        ListNode nextGroup = head;
        int count = k;
        while (count > 0 && nextGroup != null) {
            count--;
            nextGroup = nextGroup.next;
        }

        if (count > 0) return head;

        ListNode prev = ReverseKGroup(nextGroup, k);
        while (head != nextGroup) {
            ListNode temp = head.next;
            head.next = prev;
            prev = head;
            head = temp;
        }

        return prev;
    }
}
